### ISSUE TITLE
SOSConstraintList for pyomo.environ

### ISSUE THREAD

**Issue by Krabii** [Pyomo/pyomo#1856]
Is there a trivial way to create many indexed SOSConstraint() and interactively add them to a constraint container, just like it can be done with Constraint() and ConstraintList()?

according to the documentation here:
https://pyomo.readthedocs.io/en/stable/library_reference/kernel/syntax_comparison.html
there is no equivalent to the kernel syntax for that.

Is it a feature that can be done in the future or it is intentionally done this way?

Is the kernel syntax the only way to do this? If so, can the two syntax get mixed in the same ConcreteModel()?

**Comment by mrmundt** [comment]
Hello, @Krabii. The GitHub Issues are intended for bug reports and enhancement/feature requests. For help or use questions, we refer you to review our documentation for [ways to get help](https://github.com/Pyomo/pyomo#getting-help). If this is still an issue for you, please post your question on one of the preferred forums and we can engage with you there.

**Comment by Krabii** [comment]
Thanks @mrmundt,

I guess I formulated it as a question but it could be a valid feature request: "Need for a SOSConstraintList container".

Anyway, I have found a way to dynamically create SOS constraints like that:

sos_counter = 1
for i in range(N):
    setattr(model, f'sos_{sos_cnt}', SOSConstraint(var=model.my_var, index=my_list_of_indexes, sos=1))
    sos_counter += 1

It works, but it is obviously a very hacky way to achieve that. I find the ConstraintList much more readable and concise.
